### PR TITLE
Fixes compatibility issue with ruby 2.7.0 or greater

### LINKED ISSUE #15: Compatibility issues with Ruby greater or equal to 2.7
I tried using this gem with Ruby `2.7.1` and I got the following error
```
#<FrozenError: can't modify frozen String: "">
```
It seems the issue is the `http.rb` version dependency you declare in your `gemspec` file.
```
spec.add_dependency 'http', '~> 3.0'
```
[This issue seems to be fixed](https://github.com/httprb/http/issues/582) on the `4.2.1` version of `http.rb`

Maybe bumping the dependency version would be enough.

Full Stack trace
```
".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/http-3.3.0/lib/http/response/body.rb:52:in `force_encoding'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/http-3.3.0/lib/http/response/body.rb:52:in `to_s'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/bundler/gems/faraday-http-80c9bedffbb2/lib/faraday/adapter/http.rb:29:in `perform_request'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/bundler/gems/faraday-http-80c9bedffbb2/lib/faraday/adapter/http.rb:16:in `call'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday_middleware-1.0.0/lib/faraday_middleware/response_middleware.rb:36:in `call'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday_middleware-1.0.0/lib/faraday_middleware/response/follow_redirects.rb:79:in `perform_with_redirection'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday_middleware-1.0.0/lib/faraday_middleware/response/follow_redirects.rb:67:in `call'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday-1.0.1/lib/faraday/request/retry.rb:148:in `call'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday-1.0.1/lib/faraday/request/url_encoded.rb:25:in `call'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday-1.0.1/lib/faraday/rack_builder.rb:153:in `build_response'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday-1.0.1/lib/faraday/connection.rb:492:in `run_request'",
 ".rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/faraday-1.0.1/lib/faraday/connection.rb:198:in `get'",

```
Thanks

**EDIT**:
Cloned the repo locally, updated `gemspec` dependency of `http.rb` to 
` spec.add_dependency 'http', '~> 4'` specs seems to be passing. 
Have not tested further than that though.

### PR DESCRIPTION
**Pull request by hgeorgilas**
`Bumps http.rb dependency to version 4`

Resolves #15 

[Related http.rb gem issue](https://github.com/httprb/http/issues/582)

### CODE REVIEW

File: faraday-http.gemspec [hunk 1]
```diff
@@ -39,7 +39,7 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
-  spec.add_dependency 'http', '~> 3.0'
+  spec.add_dependency 'http', '~> 4'
```
Comment (iMacTia) [inline faraday-http.gemspec:42]: Never used the pessimistic operator without decimals 🤔...
I know `~> 4.0` would mean `>= 4.0`, `<5`, but `~> 4` ? Does that stop from upgrading to 5 😅?

I'd personally feel more comfortable having `~> 4.0` as I couldn't find references on how this behaves

Comment (hgeorgilas) [reply faraday-http.gemspec:42]: @iMacTia You are right, I was filling a little weird while I was just writing `4` on that `gemspec` file. The issue is fixed on `4.2.1` and the current stable release is `4.4.1`, they are bumping minor versions pretty fast. Would it be better to just lock it on `~> 4.4` ?

Comment (hgeorgilas) [reply faraday-http.gemspec]: @iMacTia I amended the commit

**Comment by olleolleolle** [comment]
Preparing to merge.

**Pull request merged**